packages/abi/src/matchers/sway-type-matchers.ts

```diff
@@ -0,0 +1,116 @@
+export type SwayType =
+  | 'void'
+  | 'bool'
+  | 'u8'
+  | 'u16'
+  | 'u32'
+  | 'u64'
+  | 'u256'
+  | 'b256'
+  | 'generic'
+  | 'string'
+  | 'stdString'
+  | 'option'
+  | 'result'
+  | 'enum'
+  | 'struct'
+  | 'b512'
+  | 'bytes'
+  | 'vector'
+  | 'tuple'
+  | 'array'
+  | 'assetId'
+  | 'evmAddress'
+  | 'rawUntypedPtr' // might not need it
+  | 'rawUntypedSlice'; // might not need it
+
+type Matcher = (type: string) => boolean;
+
+const voidMatcher: Matcher = (type) => type === '()';
+const bool: Matcher = (type) => type === 'bool';
+const u8: Matcher = (type) => type === 'u8';
+const u16: Matcher = (type) => type === 'u16';
+const u32: Matcher = (type) => type === 'u32';
+const u64: Matcher = (type) => type === 'u64';
+const u256: Matcher = (type) => type === 'u256';
+const b256: Matcher = (type) => type === 'b256';
+
+export const GENERIC_REGEX = /^generic ([^\s]+)$/m;
+const generic: Matcher = (type) => GENERIC_REGEX.test(type);
+
+export const STRING_REGEX = /str\[(?<length>[0-9]+)\]/;
+const string: Matcher = (type) => STRING_REGEX.test(type);
+
+export const TUPLE_REGEX = /^\((?<items>.+)\)$/m;
+const tuple: Matcher = (type) => TUPLE_REGEX.test(type);
+
+export const ARRAY_REGEX = /\[(?<item>[\w\s\\[\]]+);\s*(?<length>[0-9]+)\]/;
+const array: Matcher = (type) => ARRAY_REGEX.test(type);
+
+const struct: Matcher = (type) =>
```

Comment:
Adhering to the REGEX/const pattern.

```suggestion
const STRUCT_REGEX = /^struct .+$/m;
const STRUCT_STD_REGEX = /^struct std::.*(AssetId|B512|Vec|RawVec|EvmAddress|Bytes|String|RawBytes)$/m;
const struct: Matcher = (type) => STRUCT_REGEX.test(type) && !STRUCT_STD_REGEX.test(type);
```